cloud-controller-manager/do/loadbalancers.go

```diff
@@ -411,6 +451,54 @@ func (l *loadBalancers) EnsureLoadBalancerDeleted(ctx context.Context, clusterNa
 		return fmt.Errorf("failed to delete load-balancer: %s", err)
 	}
 
+	domain, err := getDomain(service)
+	if err != nil {
+		klog.Errorf("failed to fetch domain from service for cleanup: %s", err)
+		return nil
+	}
+
+	// attempt to clean up any managed A records or certificates related to LB
+	if domain != nil && lb != nil {
```

Comment:
I don't think we need to check `lb` for `nil`: it should always exists given that we check for the existence of the LB first thing in this function (and bail out if it doesn't exist).